### PR TITLE
Fix partially supported callables deprecation

### PR DESCRIPTION
**Pull request by jrmajor**
https://wiki.php.net/rfc/deprecate_partially_supported_callables

### CODE REVIEW

**Comment by jrmajor** [comment]
I was unable to run the tests locally, hoped to get some feedback from the CI, but it's not there.

**Comment by jrmajor** [comment]
Since Travis isn't working, I've moved CI to GitHub Actions. You can see it passing at https://github.com/jrmajor/Traits/commit/a3864c7c152211250cf77f9e0e735fee58005437. I've added all combinations of PHPUnit major versions and PHP versions they support, but maybe some of them can be removed.

File: dev-tools/composer.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
     "require": {
-        "php": "^7.3"
+        "php": "~8.0.0"
```
Comment (keradus) [inline dev-tools/composer.json]: why `~` and not `^` ?

Comment (jrmajor) [reply dev-tools/composer.json]: So that it does not match PHP 8.1+.

**Comment by jrmajor** [comment]
Let me know if you want me to extract the GitHub Actions setup to another PR :)

File: dev-tools/composer.json (hunk 1 above)
Comment (keradus) [reply dev-tools/composer.json]: I understand that part, but I do not understand why we don't want 8.1 here?

**Comment by keradus** [comment]
that would help @jrmajor , thanks!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -42,6 +42,9 @@ jobs:
             php-version: '8.0'
           - phpunit-version: 8
             php-version: '8.1'
+          - phpunit-version: 8
+            php-version: '8.2'
+            composer-flags: '--ignore-platform-req=PHP'
```
Comment (keradus) [inline .github/workflows/ci.yml:47]: pls don't
if we claim support for a combination of given PHP / PHPUnit versions, but only with this flag - someone in their project will not be using that flag and face issue that CI is exposing here, without this flag.

some extra details in https://github.com/php-coveralls/php-coveralls/pull/346#issuecomment-1230126135

Comment (keradus) [reply .github/workflows/ci.yml:47]: ok, it's different. we wait for phpspec/ to release new version compact with 8.2. agree to proceed

**Comment by keradus** [comment]
I will merge and release. nice work!

File: dev-tools/composer.json (hunk 1 above)
Comment (jrmajor) [reply dev-tools/composer.json]: Because we run it (and check whether it works) only on PHP 8.0.

Comment (keradus) [reply dev-tools/composer.json]: I bumped to 8.1 in #61 , green, so merged :)

Comment (jrmajor) [reply dev-tools/composer.json]: I guess it's the eternal discussion about whether we can declare compatibility with PHP 8.3 before it's released :)

**Pull request merged**